cors/cors.go

```diff
@@ -0,0 +1,127 @@
+package cors
+
+import (
+	"log"
+	"net/http"
+	"strings"
+
+	"github.com/aws/aws-lambda-go/events"
+)
+
+type OriginChecker interface {
+	Options(events.APIGatewayV2HTTPRequest) events.APIGatewayV2HTTPResponse
+	GetCorsHeaders(request events.APIGatewayV2HTTPRequest) (map[string]string, bool)
+}
+
+// Domains contains the allowed domains and the accepted methods for each one
+type Domains struct {
+	Allowed map[string]map[string]bool
+}
+
+// NewDomains returns the default Domains
+func NewOriginChecker() OriginChecker {
+	acceptedDomains := map[string]map[string]bool{
+		"thelist.app": {"DELETE": true, "GET": true, "PATCH": true, "POST": true},
+	}
+	return &Domains{Allowed: acceptedDomains}
+}
+
+func IsOptionsRequest(request events.APIGatewayV2HTTPRequest) bool {
+	return http.MethodOptions == request.RequestContext.HTTP.Method
+}
+
+// Options performs an options request
+// response details methods which are permitted to be performed from the domain in the Origin header
+func (d *Domains) Options(request events.APIGatewayV2HTTPRequest) events.APIGatewayV2HTTPResponse {
+	origin, ok := request.Headers["Origin"]
+	if !ok {
+		return events.APIGatewayV2HTTPResponse{
+			StatusCode: http.StatusBadRequest,
+			Body:       `{"error": "Origin header was not set on request"}`,
+		}
+	}
+
+	allowedMethds := d.getAllowedMethodsForOrigin(origin)
+	if len(allowedMethds) == 0 {
+		return events.APIGatewayV2HTTPResponse{
+			StatusCode: http.StatusNoContent,
+		}
+	}
+
+	return events.APIGatewayV2HTTPResponse{
+		StatusCode: http.StatusNoContent,
+		Headers: map[string]string{
+			"Access-Control-Allow-Origin":  origin,
+			"Access-Control-Allow-Methods": commaSeperateTrueKeys(allowedMethds),
+		},
```

Comment:
We also need `"Access-Control-Allow-Headers": "content-type"` as:

> When using the Content-Type header, only the following values are allowed: application/x-www-form-urlencoded, multipart/form-data, or text/plain[[ref]](https://auth0.com/blog/cors-tutorial-a-guide-to-cross-origin-resource-sharing/)

i.e. any other header value needs to be allowed